### ISSUE TITLE
Make _SignalReactorMixin a newstyle class

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4287]
|[<img alt="fijal's avatar" src="https://avatars.githubusercontent.com/u/85944?s=50" width="50" height="50">](https://github.com/fijal)| @fijal reported|
|-|-|
|Trac ID|trac#4287|
|Type|enhancement|
|Created|2010-02-21 19:35:53Z|
|Branch|https://github.com/twisted/twisted/tree/reactor-newstyle-mixin-4287|

This greatly speeds up iteration benchmark on top of pypy-c.

reactors are new-style classes anyway, but they have object stacked later on to their inheritance hierarchy. A single run (no science) of iteration benchmark https://code.launchpad.net/~exarkun/+junk/twisted-benchmarks:

reactor as old style - 133419.2 req/s
reactor as new style - 271452.4 req/s

<details><summary>Searchable metadata</summary>

```
trac-id__4287 4287
type__enhancement enhancement
reporter__fijal fijal
priority__normal normal
milestone__ 
branch__branches_reactor_newstyle_mixin_4287 branches/reactor-newstyle-mixin-4287
branch_author__fijal fijal
status__closed closed
resolution__fixed fixed
component__core core
keywords__pypy pypy
time__1266780953000000 1266780953000000
changetime__1266872004000000 1266872004000000
version__None None
owner__fijal fijal

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="fijal's avatar" src="https://avatars.githubusercontent.com/u/85944?s=50" width="50" height="50">](https://github.com/fijal)<a name="note_1"></a>|@fijal commented|
|-|-|

(In [28320]) Create a branch for making reactor mixins a new style class

**Comment by twisted-trac** [comment]
|[<img alt="fijal's avatar" src="https://avatars.githubusercontent.com/u/85944?s=50" width="50" height="50">](https://github.com/fijal)|@fijal removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="jkakar's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|jkakar set owner to @fijal|
|-|-|

[1]

+class ObjectModelIntegrationTest(ReactorBuilder):
+    """ Checks that all reactors on a platform have mro
+    containing only new style classes
+    """

I believe this should be formatted as follows:

class ObjectModelIntegrationTest(ReactorBuilder):
    """
    Checks that all reactors on a platform have mro containing only
    new style classes.
    """


[2]

Please add a docstring to test_newstyleReactor.


Small branches rock.

**Comment by twisted-trac** [comment]
|[<img alt="fijal's avatar" src="https://avatars.githubusercontent.com/u/85944?s=50" width="50" height="50">](https://github.com/fijal)|@fijal removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="djfroofy's avatar" src="https://avatars.githubusercontent.com/u/89760?s=50" width="50" height="50">](https://github.com/djfroofy)|@djfroofy set owner to @djfroofy<br>@djfroofy set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="fijal's avatar" src="https://avatars.githubusercontent.com/u/85944?s=50" width="50" height="50">](https://github.com/fijal)|@fijal removed owner<br>@fijal set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve set owner to @fijal|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_8"></a>|@therve commented|
|-|-|

Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_9"></a>|@therve set status to `closed`|
|-|-|

(In [28410]) Merge reactor-newstyle-mixin-4287

Author: fijal
Reviewers: jkakar, therve
Fixes [#4287](https://github.com/twisted/twisted/issues/4287)

Make _SignalReactorMixin a newstyle class to speedup pypy.

**Comment by twisted-trac** [comment]
|[<img alt="djfroofy's avatar" src="https://avatars.githubusercontent.com/u/89760?s=50" width="50" height="50">](https://github.com/djfroofy)<a name="note_10"></a>|@djfroofy commented|
|-|-|

Merged cleanly and all unit tests passing.  Also ran iteration.py benchmark with CPython and don't see much difference (only slightly faster):

Twisted trunk
53868.8 req/sec (269344 iterations in 5 seconds)
52836.0 req/sec (264180 iterations in 5 seconds)
53602.0 req/sec (268010 iterations in 5 seconds)

Merged 4287
54096.0 req/sec (270480 iterations in 5 seconds)
53469.6 req/sec (267348 iterations in 5 seconds)
53799.4 req/sec (268997 iterations in 5 seconds)

Also, considering reactor is new style class already, I can't see any possible backwards compat issues.
